javascript/react/src/components/Codebook/MappingFields.tsx

```diff
@@ -0,0 +1,152 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type { Field } from '@datashaper/schema'
+import { ActionButton, Checkbox, Label, TextField } from '@fluentui/react'
+import { memo, useCallback, useEffect, useState } from 'react'
+
+import { useTextChange } from './CodebookFieldEditor.hooks.js'
+import type { CodebookFieldEditorProps } from './CodebookFieldEditor.types.js'
+import {
+	useChangeMappingDisplay,
+	useChangeMappingNull,
+	useChangeMappingValue,
+} from './MappingFields.hooks.js'
+import {
+	FieldLabel,
+	FlexColumn,
+	MappingContainer,
+} from './MappingFields.styles.js'
+interface NewValueField {
+	value: string
+	display: string
+}
+export const MappingFields: React.FC<CodebookFieldEditorProps> = memo(
+	function MappingFields({ field, showInlineLabel, onChange }) {
+		const [values, setValues] = useState(field.mapping)
```

Comment:
Why do we need the internal state here?

Reply:
Different from recode, in here I always have at least one field and then I can fill it and click add, that adds the item to the table but we continue to show an empty field for the next one. But I will change it to be like recode where we only show the new field when clicked, then it will already be a part of the list anyway.